hugegraph-api/src/main/java/com/baidu/hugegraph/api/graph/UpdateStrategy.java

```diff
@@ -0,0 +1,181 @@
+            E.checkState((oldProperty instanceof Date ||
+                          oldProperty instanceof Long) &&
+                         (newProperty instanceof Date ||
+                          newProperty instanceof Long),
+                         getFormat(oldProperty, newProperty, "Date or Long"));
+        }
+    },
+
+    // TODO: Support Number compare later
+    SMALLER {
+        @Override
+        Object updatePropertyValue(Object oldProperty, Object newProperty) {
+            return getComparedDate(oldProperty, newProperty, SMALLER);
+        }
+
+        @Override
+        void checkPropertyType(Object oldProperty, Object newProperty) {
+            BIGGER.checkPropertyType(oldProperty, newProperty);
+        }
+    },
+
+    APPEND {
+        @Override
+        Object updatePropertyValue(Object oldProperty, Object newProperty) {
+            return ((List) oldProperty).addAll((List) newProperty);
+        }
+
+        @Override
+        void checkPropertyType(Object oldProperty, Object newProperty) {
+            E.checkState(oldProperty instanceof List &&
+                         newProperty instanceof List,
+                         getFormat(oldProperty, newProperty, "List"));
+        }
+    },
+
+    UNION {
+        @Override
+        Object updatePropertyValue(Object oldProperty, Object newProperty) {
+            return getComparedSet(oldProperty, newProperty, UNION);
+        }
+
+        @Override
+        void checkPropertyType(Object oldProperty, Object newProperty) {
+            // JsonElements are always List-type, so allows two type now.
+            E.checkState((oldProperty instanceof Set ||
+                          oldProperty instanceof List) &&
+                         (newProperty instanceof Set ||
+                          newProperty instanceof List),
+                         getFormat(oldProperty, newProperty, "Set or List"));
+        }
+    },
+
+    INTERSECTION {
+        @Override
+        Object updatePropertyValue(Object oldProperty, Object newProperty) {
+            return getComparedSet(oldProperty, newProperty, INTERSECTION);
+        }
+
+        @Override
+        void checkPropertyType(Object oldProperty, Object newProperty) {
+            UNION.checkPropertyType(oldProperty, newProperty);
+        }
+    },
+
+    // TODO: Merge from "action", how to handle List? (have same elements)
+    ELIMINATE {
+        @Override
+        Object updatePropertyValue(Object oldProperty, Object newProperty) {
+            return null;
+        }
+
+        @Override
+        void checkPropertyType(Object oldProperty, Object newProperty) {
+            throw new NotSupportException("UpdateStrategy : eliminate");
+        }
+    };
+
+    abstract Object updatePropertyValue(Object oldProperty, Object newProperty);
+
+    abstract void checkPropertyType(Object oldProperty, Object newProperty);
+
+    public static Object checkAndUpdateProperty(@NotNull Object oldProperty,
```

Comment:
we can use member method here instead of static method

Reply:
but how to use non-static method in other class directly, like `UpdateStrategy.func()` 
unless using `UpdateStrategy.SUM.func()`? (and enum is not stable)

Reply:
we can call its method like common class:
```
UpdateStrategy strategy = UpdateStrategy.valueOf(...)
strategy.func()
```